src/pip/_internal/req/constructors.py

```diff
@@ -200,6 +200,59 @@ def install_req_from_editable(
     )
 
 
+def _looks_like_path(name):
+    # type: (str) -> bool
+    """Checks whether the string "looks like" a path on the filesystem.
+
+    This does not check whether the target actually exists, only judge from the
+    appearance. Returns true if any of the following is true:
+
+    * A path separator is found (either os.path.sep or os.path.altsep).
+    * The string starts with "." (current directory).
+    """
+    if os.path.sep in name:
+        return True
+    if os.path.altsep is not None and os.path.altsep in name:
+        return True
+    if name.startswith('.'):
+        return True
```

Comment:
I just realised this if does not seem to work as intended. A `./whatever` string would’ve been caught in previous checks. This only matters for strings like `.whatever`, which I guess still does look like a path…? (but then the docstring is not accurate)

Reply:
@uranusjr that's exactly it. I don't really know why some package would start with `.`, but I'll add a test case for it and add it to the docstring.

Reply:
Packages *can’t* start with a dot, so this check doesn’t really matter either way :p But it’s better to remove it since its mere existence can be confusing to future readers.

Reply:
Nice. I have removed it then.

Reply:
@uranusjr oops, actually you can use `.` to install a package. You can use it to install the current directory as a package if it has a `setup.py` file.

Reply:
@vinicyusmacedo Ah that makes sense, I missed that case! I guess it’s better to keep it (or rewrite it to `if name == "."`) then. You’ll still need to add a sentence to the docstring to decribe the intent though; future readers would wonder.

Reply:
@uranusjr added the `==` instead of `startswith`

Reply:
Though I still have a problem: the Windows tests might fail with the `==` since the path separator is different. I'll go with `name.startswith` then (I could make separate test cases for Windows, but that doesn't sound so good).

Reply:
I believe the sep and altsep parts cover the different separators (if my memory of implementation from other projects serves).

Reply:
From what I understood from the docs, it appears to be only available on Windows (the altsep on Windows would be the forward-slash)

https://docs.python.org/3/library/os.html#os.altsep

Reply:
That is correct, hence the first test would detect `\` on Windows and `/` on POSIX; the second test detects `/` on Windows (always false on POSIX).

You could add a simple Windows-only test like this, if you’re inclined to:

```python
@pytest.mark.parametrize('path', [
     '.\\path\\to\\installable',
     'relative\\path',
     'C:\\absolute\\path',
 ])
@pytest.skipif(os.path.altsep != '\\')
 def test_looks_like_path_win(path):
     assert _looks_like_path(path) == True
```

Reply:
@uranusjr I could also skip this test if `not sys.platform.startswith("win")`